internal/controller/pipelinerollout_controller.go

```diff
@@ -503,13 +531,21 @@ func pipelineNeedsUpdating(ctx context.Context, newPipelineDef *kubernetes.Gener
 	return !pipelineSpecsEqual, nil
 }
 
-func pipelineIsUpdating(newPipelineDef *kubernetes.GenericObject, existingPipelineDef *kubernetes.GenericObject) (bool, error) {
+// return true if Pipeline (or its children) is still in the process of being reconciled
+func pipelineIsUpdating(pipelineRollout *apiv1.PipelineRollout, newPipelineDef *kubernetes.GenericObject, existingPipelineDef *kubernetes.GenericObject) (bool, error) {
 	existingPipelineStatus, err := kubernetes.ParseStatus(existingPipelineDef)
 	if err != nil {
 		return false, err
 	}
-	// TODO: will need to update this to account for Vertex ObservedGeneration once that's ready
-	return !pipelineObservedGenerationCurrent(newPipelineDef.Generation, existingPipelineStatus.ObservedGeneration), nil
+	// if Pipeline's ObservedGeneration is old, then Numaflow Controller hasn't even seen the generation change yet
+	if !pipelineObservedGenerationCurrent(newPipelineDef.Generation, existingPipelineStatus.ObservedGeneration) {
+		return true, nil
+	}
+
+	// note if Pipeline's children are still being updated
+	_, pipelineChildResourceReason := getPipelineChildResourceHealth(pipelineRollout.Status.Conditions)
+
+	return pipelineChildResourceReason == "Progressing", nil
```

Comment:
Here I incorporated the new `Pipeline.Status.ChildResourcesHealth==Progressing` to know if the Pipeline's children are still in the process of being reconciled, now that it is available. 

Note that `numaflow-rc:0.6.0` and `numaflow-rc:0.7.0` both set this